app/services/frontend_logger.rb

```diff
@@ -1,27 +1,45 @@
 class FrontendLogger
   attr_reader :analytics, :event_map
 
+  # @param [Analytics] analytics
+  # @param [Hash{String=>Symbol,#call}] event_map map of string event names to method names
+  #   on Analytics, or a custom implementation that's callable (like a Proc or Method)
   def initialize(analytics:, event_map:)
     @analytics = analytics
     @event_map = event_map
   end
 
+  # Logs an event and converts the payload to the correct keyword args
+  # @param [String] name
+  # @param [Hash<String,Object>] attributes payload with string keys
   def track_event(name, attributes)
-    if (analytics_method = event_map[name])
-      analytics.send(analytics_method.name, **hash_from_method_kwargs(attributes, analytics_method))
+    analytics_method = event_map[name]
+
+    if analytics_method.is_a?(Symbol)
+      analytics.send(
+        analytics_method,
+        **hash_from_kwargs(attributes, AnalyticsEvents.instance_method(analytics_method)),
```

Comment:
9e28cb858f018dd10f695f7a918174776d079303 fixes an issue with some specific events in IdV not working. I think this is also what `transform_values` was helping with previously, since trying to get the method signature from the `analytics` instance is going to have issues with how `IdV::AnalyticsEventsEnhancer` overrides the method and loses the signature.

```
| [3] pry(#<FrontendLogger>)> analytics.method(analytics_method)
=> #<Method: Analytics(Idv::AnalyticsEventsEnhancer)#idv_personal_key_acknowledgment_toggled(**kwargs) /identity-idp/app/services/idv/analytics_events_enhancer.rb:39>
```

https://github.com/18F/identity-idp/blob/9d5698a0c75a694618e94b5a0ebb88a84d4dad74/app/services/idv/analytics_events_enhancer.rb#L39

We could also consider moving this back into `transform_values`, and maybe there's even something where we can consolidate the handling so that the `EVENT_MAP` hash values are _all_ the same type of callable thing? The issue comes with binding though... maybe we can have a small logic branch to normalize an unbound method to bind it with `analytics`?

```rb
analytics_method = analytics_method.bind(analytics) if analytics_method.is_a?(UnboundMethod)
analytics_method.respond_to?(:call)
  analytics_method.call(**hash_from_kwargs(attributes, analytics_method))
else
  analytics.track_event("Frontend: #{name}", attributes)
end
```

Reply:
I feel good! 🚢